patch-db-macro-internals/src/lib.rs

```diff
@@ -101,159 +101,164 @@ fn build_model_struct(
-                                &heck::ShoutyKebabCase::to_shouty_kebab_case(
-                                    ident.to_string().as_str(),
-                                ),
-                                ident.span(),
-                            ),
-                            _ => LitStr::new(&ident.to_string(), ident.span()),
-                        },
-                    ));
+                            ident.span(),
+                        ),
+                        _ => LitStr::new(&ident.to_string(), ident.span()),
+                    }));
                 }
             }
         }
-        Fields::Unnamed(f) => {
-            if f.unnamed.len() == 1 {
-                // newtype wrapper
-                let field = &f.unnamed[0];
+        Fields::Unnamed(f) if f.unnamed.len() == 1 => {
+            // newtype wrapper
+            let field = &f.unnamed[0];
+            let ty = &field.ty;
+            let inner_model: Type = if let Some(child_model_name) = field
+                .attrs
+                .iter()
+                .filter(|attr| attr.path.is_ident("model"))
+                .map(|attr| attr.parse_args::<MetaNameValue>().unwrap())
+                .filter(|nv| nv.path.is_ident("name"))
+                .find_map(|nv| match nv.lit {
+                    Lit::Str(s) => Some(s),
+                    _ => None,
+                }) {
+                let child_model_ty = Ident::new(&child_model_name.value(), child_model_name.span());
+                syn::parse2(quote! { #child_model_ty }).unwrap()
+            } else if field.attrs.iter().any(|attr| attr.path.is_ident("model")) {
+                syn::parse2(quote! { <#ty as patch_db::HasModel>::Model }).unwrap()
+            } else {
+                syn::parse2(quote! { patch_db::Model::<#ty> }).unwrap()
+            };
+            return quote! {
+                #[derive(Debug)]
+                #model_vis struct #model_name(#inner_model);
+                impl std::clone::Clone for #model_name {
+                    fn clone(&self) -> Self {
+                        #model_name(self.0.clone())
+                    }
+                }
+                impl core::ops::Deref for #model_name {
+                    type Target = #inner_model;
+                    fn deref(&self) -> &Self::Target {
+                        &self.0
+                    }
+                }
+                impl core::ops::DerefMut for #model_name {
+                    fn deref_mut(&mut self) -> &mut Self::Target {
+                        &mut self.0
+                    }
+                }
+                impl From<patch_db::json_ptr::JsonPointer> for #model_name {
+                    fn from(ptr: patch_db::json_ptr::JsonPointer) -> Self {
+                        #model_name(#inner_model::from(ptr))
+                    }
+                }
+                impl From<patch_db::ModelPaths> for #model_name {
```

Comment:
Needing to create a model from a path, there are some into()/ from that convert types from optional and such